File: examples/go/linked_list_1.go
package main

import "fmt"

type Node struct {
	val  int
	next *Node
}

func (n *Node) append(v int) *Node {
	if n.next == nil {
		n.next = &Node{val: v}
		return n.next
	}
	return n.next.append(v)
}

func (n *Node) print() {
	if n == nil {
		return
	}
	fmt.Println(n.val)
	n.next.print()
}

func main() {
	head := &Node{val: 1}
	head.append(2).append(3)
	head.print()
}

File: examples/java/LinkedList1.java
public class LinkedList1 {
    static class Node {
        int data;
        Node next;
        Node(int d) { data = d; next = null; }
    }

    static Node head;

    static void insertEnd(int val) {
        Node n = new Node(val);
        if (head == null) { head = n; return; }
        Node curr = head;
        while (curr.next != null) curr = curr.next;
        curr.next = n;
    }

    static void print() {
        Node curr = head;
        while (curr != null) {
            System.out.print(curr.data + (curr.next != null ? " -> " : " -> null\n"));
            curr = curr.next;
        }
    }

    public static void main(String[] args) {
        insertEnd(10);
        insertEnd(20);
        insertEnd(30);
        print();
    }
}
